### PR TITLE
Chore: Improve typings and ignore `Graph`

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- improves some typings to avoid `any`/type assertions
- ignores `public/app/plugins/panel/graph` from the eslint list since it's going to be removed with angular

**Why do we need this feature?**

- 🧹 

**Who is this feature for?**

- grafana devs

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-data/src/types/live.ts [hunk 1]
```diff
@@ -137,8 +137,8 @@ export type LiveChannelId = string;
  */
 export interface LiveChannelAddress {
   scope: LiveChannelScope;
-  namespace: string; // depends on the scope
-  path: string;
+  namespace: string | undefined; // depends on the scope
+  path: string | undefined;
```
Comment (joshhunt) [inline packages/grafana-data/src/types/live.ts:141]: Sounds like a good use for tagged union. but this is fine also 👍🏻

Comment (ryantxu) [reply packages/grafana-data/src/types/live.ts:141]: hymm -- i think this should stay the same (a valid address everything is set), but the editor should use Partial<LiveChannelAddress>

File: public/app/plugins/panel/canvas/editor/element/elementEditor.tsx [hunk 2]
```diff
@@ -36,10 +36,10 @@ export function getElementEditor(opts: CanvasEditorOptions): NestedPanelOptions<
 
     // Note that canvas editor writes things to the scene!
     values: (parent: NestedValueAccess) => ({
-      getValue: (path: string) => {
+      getValue: (path) => {
         return lodashGet(opts.element.options, path);
       },
-      onChange: (path: string, value: any) => {
+      onChange: (path, value) => {
```
Comment (tskarhed) [inline public/app/plugins/panel/canvas/editor/element/elementEditor.tsx:42]: In these cases, is it that it implicitly already has a non-`any`, but it has been overridden by `any`?

**Review by joshhunt — APPROVED** [review]

File: public/app/plugins/panel/canvas/editor/element/elementEditor.tsx (hunk 2 above)
Comment (ashharrison90) [reply public/app/plugins/panel/canvas/editor/element/elementEditor.tsx:42]: implicitly it's still `any` (you can see the definition for `onChange` [here](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/utils/OptionsUIBuilders.ts#L165)), so this doesn't really improve the typing so much as decrease the overall count of `any`s by 1.

**but** i still think that's important. it's very hard to be motivated to fix these type errors properly when there's 2000 remaining. once we get down to like 100 then there's light at the end of the tunnel.

**Pull request merged**